include/openenclave/bits/sgx/sgxproperties.h

```diff
@@ -84,24 +88,33 @@ typedef struct _oe_sgx_enclave_properties
  * an enclave binary. These properties can be overwritten at sign time by
  * the oesign tool.
  *
- * @param PRODUCT_ID ISV assigned Product ID (ISVPRODID) to use in the
+ * @param[in] PRODUCT_ID ISV assigned Product ID (ISVPRODID) to use in the
  * enclave signature
- * @param SECURITY_VERSION ISV assigned Security Version number (ISVSVN)
+ * @param[in] SECURITY_VERSION ISV assigned Security Version number (ISVSVN)
  * to use in the enclave signature
- * @param ALLOW_DEBUG If true, allows the enclave to be created with
+ * @param[in] EXTENDED_PRODUCT_ID ISV assigned Extended Product ID (ISVEXTPRODID)
+ *  to use in the enclave signature
+ * @param[in] FAMILY_ID ISV assigned Product Family ID (ISVFAMILYID)
+ * to use in the enclave signature
+ * @param[in] ALLOW_DEBUG If true, allows the enclave to be created with
  * OE_ENCLAVE_FLAG_DEBUG and debugged at runtime
- * @param HEAP_PAGE_COUNT Number of heap pages to allocate in the enclave
- * @param STACK_PAGE_COUNT Number of stack pages per thread to reserve in
+ * @param[in] ALLOW_KSS If true, allows the enclave to be created with
+ * kss properties
+ * @param[in] HEAP_PAGE_COUNT Number of heap pages to allocate in the enclave
+ * @param[in] STACK_PAGE_COUNT Number of stack pages per thread to reserve in
  * the enclave
- * @param TCS_COUNT Number of concurrent threads in an enclave to support
+ * @param[in] TCS_COUNT Number of concurrent threads in an enclave to support
  */
 // Note: disable clang-format since it badly misformats this macro
 // clang-format off
 
-#define OE_SET_ENCLAVE_SGX(                                               \
+#define _OE_SET_ENCLAVE_SGX_IMPL(                                         \
     PRODUCT_ID,                                                           \
     SECURITY_VERSION,                                                     \
+    EXTENDED_PRODUCT_ID,                                                  \
+    FAMILY_ID,                                                            \
     ALLOW_DEBUG,                                                          \
+    ALLOW_KSS,                                                            \
```

Comment:
By the current design, this flag is "REQUIRE_KSS" instead of "ALLOW_KSS". When this flag is set, if the underlining system does not support KSS, enclave loading will fail. Therefore, KSS is "REQUIRED". In future update, with CONFIGID/CONFIGSVN support, a developer who does not use non-zero FAMILY_ID.EXTENDED_PRODUCT_ID might implement an enclave that utilizes CONFIG_ID/CONFIG_SVN feature when KSS is available, but still functions without CONFIG_ID/CONFIG_SVN. In that case, the developer should indicate to the enclave loader that KSS is "DESIRED". I believe defining the flag as "REQUIRE_KSS" still works in that scenario. The zero value on this flag means "NOT_REQUIRED", no conflict with "DESIRED".

Reply:
Updated to REQUIRE_KSS, thanks